src/partners/godex.ts

```diff
@@ -0,0 +1,114 @@
+import { asArray, asObject, asOptional, asString, asUnknown } from 'cleaners'
+import fetch from 'node-fetch'
+
+import { PartnerPlugin, PluginParams, PluginResult, StandardTx } from '../types'
+
+// CLEANER that verfies the data fetched matches the given format and data types
+const asGodexTx = asObject({
+  status: asString,
+  hash_in: asOptional(asString),
+  deposit: asString,
+  coin_from: asString,
+  deposit_amount: asString,
+  withdrawal: asString,
+  coin_to: asString,
+  withdrawal_amount: asString,
+  created_at: asString // Date and time when transaction was created
+})
+
+const asGodexResult = asArray(asUnknown)
+
+const QUERY_LOOKBACK = 1000 * 60 * 60 * 24 * 20 // 20 days
+
+export async function queryGodex(
+  pluginParams: PluginParams
+): Promise<PluginResult> {
+  const ssFormatTxs: StandardTx[] = []
+  const limit = 100
+  let apiKey
+  let offset = 0
+  let lastCheckedTimestamp
+
+  if (typeof pluginParams.settings.latestTimeStamp !== 'number') {
```

Comment:
you should probably say === number first so that the default value is set if it doesnt succeed